tests/test_utils.py

```diff
@@ -35,6 +36,85 @@ def test_buildconfig_update():
     assert x == {'a': 'A', 'strategy': {'b1': 'newB1', 'b3': 'B3', 'b11': {}}, 'c': 'C', 'd': 'D'}
 
 
+def has_ist(buildconfig):
```

Comment:
What if this was `has_trigger(trigger_type)` ?

```python
def has_trigger(buildconfig, trigger_type):
...
    if t.get('type') == trigger_type):
        return True
...
```

I think it'll make it easier to read the tests:
```python
has_trigger(orig, 'ImageChange')
has_trigger(orig, 'Generic')
```